src/Illuminate/Contracts/Routing/UrlGenerator.php

```diff
@@ -76,4 +76,18 @@ public function action($action, $parameters = [], $absolute = true);
      * @return $this
      */
     public function setRootControllerNamespace($rootNamespace);
+
+    /**
+     * Get the request instance.
+     *
+     * @return \Illuminate\Http\Request
+     */
+    public function getRequest();
```

Comment:
I feel like we're exposing the Http component too much here. Any chance we can solve this better in the implementation of the `guest` method?

Reply:
No idea how to else solve it right now. Besides the `guest` method the `refresh` and `createRedirect` methods are also using the request object provided by `getRequest` from the UrlGenerator.

Since the request is required for functionality in at least the `guest` and `createRedirect` (here the whole request object is needed for the created RedirectResponse) methods the only solution I can think of is to give the Redirector an own request reference...But I don't think this solution is any better than the one already in use.

Reply:
Just make `__construct` want a request, instead of calling getRequest inside it?

Reply:
I don't see why an UrlGenerator implementation should be forced to have a `getRequest` method. Not only do I think that this method shouldn't be added to the contract but also other methods should be reevaluated for Laravel 7 if they should still be on the contract. For example the `setRootControllerNamespace` method which is IMO just an implementation detail and I don't see why that is on the contract. If that information is needed for the implementation of the `action` method which is also on the contract then the implementation should get that information through the constructor.

Reply:
I see your point @X-Coder264. I'm gonna have a look on it again.